### PR TITLE
Add Spatial Transformer Networks Tutorial

### PR DESCRIPTION
**Pull request by GHamrouni**
Add Spatial Transformer Networks Tutorial to the beginner section. 
The tutorial documents the use of the newly added modules **affine_grid** and **grid_sample**.

### CODE REVIEW

File: beginner_source/spatial_transformer_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+#
+# In this post we experiment with the classic MNIST dataset. Using a 
+# standard convolutional network augmented with a spatial transformer 
+# network.
+
+use_cuda = torch.cuda.is_available()
+
+# Training dataset
+train_loader = torch.utils.data.DataLoader(
+    datasets.MNIST('../data', train=True, download=True,
```
Comment (chsasank) [inline beginner_source/spatial_transformer_tutorial.py]: Do not use `../data`.
Rather download to current dir.
Something like 
```
train_dataset = datasets.MNIST(root='.', ...)
train_dataset.download()
train_loader = DataLoader(train_dataset, ...)
```

**Comment by chsasank** [comment]
Hey thanks for the tutorial!
I haven't read the whole thing yet, but it looks great from what I saw. I will give my feedback later.

Don't you think this should be in intermediate rather than beginner?

**Comment by GHamrouni** [comment]
Thanks for the feedback :)
I'm agree, maybe it a little bit advanced for the beginner section.

**Comment by GHamrouni** [comment]
Hey,
I moved the tutorial to intermediate and changed the download dir as you suggested :)

**Comment by GHamrouni** [comment]
Please let me know if you have further feedbacks.

**Comment by GHamrouni** [comment]
Hello,

I fixed the ToC rendering for STN tutorial. And fixed all the typos.

**Comment by chsasank** [comment]
Hey, your tutorial looks fine overall. I've added a few comments in the review.

**Comment by GHamrouni** [comment]
Thank you very much for the feedback, I will clear all flake8 and fix pep8 warnings. I tried reducing the epochs, we obtain more or less the same output using 20-30 iterations.

**Comment by GHamrouni** [comment]
Hey

I fixed all linter warnings and reducing the overall running time.
I think that the tutorial is in good shape for merging, please let me know if you have further feedback.
Thank you.

**Comment by chsasank** [comment]
Great, this works! I

**Comment by chsasank** [comment]
Thanks for taking the effort to write a tutorial!  😃

**Pull request merged**